pyfr/solvers/aceuler/kernels/rsolvers/rusanov.mako

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+<%namespace module='pyfr.backends.base.makoutil' name='pyfr'/>
+<%include file='pyfr.solvers.aceuler.kernels.flux'/>
+
+<%pyfr:macro name='rsolve' params='ul, ur, n, nf'>
+    // Compute the left and right fluxes + velocities and pressures
+    fpdtype_t fl[${ndims}][${nvars}], fr[${ndims}][${nvars}];
+    fpdtype_t vl[${ndims}], vr[${ndims}];
+
+    ${pyfr.expand('inviscid_flux', 'ul', 'fl')};
+    ${pyfr.expand('inviscid_flux', 'ur', 'fr')};
+
+% for i in range(ndims):
+    vl[${i}] = ul[${i + 1}];
```

Comment:
``pyfr.array`` may help you with initalising these critters and avoid the need for an explicit for loop.